doc/feature/synapses.md

```diff
@@ -0,0 +1,232 @@
+Synapse support in Brain
+============
+
+The low-level data access for synapses provided by Brion needs a complementary
+API in Brain to support various common higher-level uses cases, like semantic
+synapse object access or full-circuit synapse density voxelization, to name the
+two extremes.
+
+## Requirements
+
+* Semantic synapse objects that give read-access to all properties
+* Array access which allows vectorization
+* Filtering of synapses for pathway use case (projected synapses of two targets)
+* Lazy-loading to support out-of-core scenarios
+* User-transparent KVS caching
+
+## New dependencies
+
+None
+
+## API
+
+```cpp
+
+namespace brain
+{
+    /*
+     * Synapse GID.
+     *
+     * The GID of a synapse is the a tuple of two numbers:
+     * - The GID of the post-synaptic cell.
+     * - The index of the synapse in the array of afferent contacts
+     *   of the post-synaptic cell before pruning/filtering.
+     * GIDs are invariant regardless of how the structural touches are
+     * converted into functional synapses during circuit building.
+     */
+    typedef std::pair< uint32_t, size > SynapseGID;
+
+    // loading of data during SynapseStream.read(), otherwise load happens on
+    // first touch
```

Comment:
pre and post GIDs are always loaded by SynapseStream.read().

This needs to be more specific about what is a position attribute (because yes, they are also attributes), for example, postSectionGID is an attribute or a position (it's a topological position)?